src/snowflake/snowpark/context.py

```diff
@@ -21,6 +21,10 @@
 _should_continue_registration: Optional[Callable[..., bool]] = None
 
 
+# Global flag that determines if structured type semantics should be used
+_should_use_structured_type_semantics = False
```

Comment:
who's responsible for setting this parameter, also do we need server side parameter for this?

Reply:
This should not be enabled until structured type support is rolled out. There will eventually be a server side parameter that we can track instead of this flag.

Reply:
cool, if this is not rolled out yet then I'm good with this module var